addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/Connection.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.icloud.internal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Base64;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * Handles communication with the apple server. Provides methods to
+ * get device information and to find a device.
+ *
+ * @author Patrik Gfeller - Initial Contribution
+ *
+ */
+public class Connection {
+    private final Logger logger = LoggerFactory.getLogger(Connection.class);
+    private final String iCloudURL = "https://www.icloud.com";
+    private final String iCloudApiURL = "https://fmipmobile.icloud.com/fmipservice/device/";
+    private final String iCloudAPIRequestDataCommand = "/initClient";
+    private final String iCloudAPIPingDeviceCommand = "/playSound";
+    private final Gson gson = new GsonBuilder().create();
+    private final String dataRequest = gson.toJson(ICloudDataRequest.defaultInstance());
+
+    private final byte[] authorization;
+    private URL iCloudDataRequestURL;
+    private URL iCloudFindMyDeviceURL;
+
+    public Connection(String appleId, String password) throws MalformedURLException {
+        authorization = Base64.getEncoder().encode((appleId + ":" + password).getBytes());
+        iCloudDataRequestURL = new URL(iCloudApiURL + appleId + iCloudAPIRequestDataCommand);
+        iCloudFindMyDeviceURL = new URL(iCloudApiURL + appleId + iCloudAPIPingDeviceCommand);
+    }
+
+    public String requestDeviceStatusJSON() throws IOException {
+        HttpsURLConnection connection = connect(iCloudDataRequestURL);
+        String response = postRequest(connection, dataRequest);
+        connection.disconnect();
+
+        return response;
+    }
+
+    /***
+     * Sends a "find my device" request.
+     *
+     * @throws IOException
+     */
+    public void findMyDevice(String id) throws IOException {
+        String request = "{ \n \"device\": \"" + id + "\", \n \"subject\": \"Find My Device alert\" \n }";
```

Comment:
Can you or will you create a follow up PR to convert this to real object? I'm okay with leaving it here for this release.

Reply:
=> https://github.com/openhab/openhab2-addons/pull/2975